es/es-repository-azure/src/main/java/org/elasticsearch/repositories/azure/AzureStorageSettings.java

```diff
@@ -0,0 +1,245 @@
+    }
+
+    @Override
+    public String toString() {
+        final StringBuilder sb = new StringBuilder("AzureStorageSettings{");
+        sb.append("account='").append(account).append('\'');
+        sb.append(", key='").append(key).append('\'');
+        sb.append(", timeout=").append(timeout);
+        sb.append(", endpointSuffix='").append(endpointSuffix).append('\'');
+        sb.append(", maxRetries=").append(maxRetries);
+        sb.append(", proxy=").append(proxy);
+        sb.append(", locationMode='").append(locationMode).append('\'');
+        sb.append('}');
+        return sb.toString();
+    }
+
+    /**
+     * Parse and read all settings available under the azure.client.* namespace
+     * @param settings settings to parse
+     * @return All the named configurations
+     */
+    public static Map<String, AzureStorageSettings> load(Settings settings) {
+        // Get the list of existing named configurations
+        final Map<String, AzureStorageSettings> storageSettings = new HashMap<>();
+        for (final String clientName : ACCOUNT_SETTING.getNamespaces(settings)) {
+            storageSettings.put(clientName, getClientSettings(settings, clientName));
+        }
+        if (false == storageSettings.containsKey("default")) {
+            // in case no setting named "default" has been set, let's define our "default"
+            // as the first named config we get
+            storageSettings.put("default", getClientSettings(settings, "default"));
+        }
+        assert storageSettings.containsKey("default") || storageSettings.isEmpty() : "always have 'default' if any";
+        return Collections.unmodifiableMap(storageSettings);
+    }
+
+    // pkg private for tests
+    /** Parse settings for a single client. */
+    static AzureStorageSettings getClientSettings(Settings settings, String clientName) {
+        try (SecureString account = getConfigValue(settings, clientName, ACCOUNT_SETTING);
+             SecureString key = getConfigValue(settings, clientName, KEY_SETTING)) {
+            return new AzureStorageSettings(account.toString(), key.toString(),
+                getValue(settings, clientName, ENDPOINT_SUFFIX_SETTING),
+                getValue(settings, clientName, TIMEOUT_SETTING),
+                getValue(settings, clientName, MAX_RETRIES_SETTING),
+                getValue(settings, clientName, PROXY_TYPE_SETTING),
+                getValue(settings, clientName, PROXY_HOST_SETTING),
+                getValue(settings, clientName, PROXY_PORT_SETTING));
+        }
+    }
+
+    private static <T> T getConfigValue(Settings settings, String clientName,
+                                        Setting.AffixSetting<T> clientSetting) {
+        final Setting<T> concreteSetting = clientSetting.getConcreteSettingForNamespace(clientName);
+        return concreteSetting.get(settings);
+    }
+
+    public static <T> T getValue(Settings settings, String groupName, Setting<T> setting) {
```

Comment:
I think `getConfigValue` can be used instead of `getValue` but just leaving this as a side-note as this whole name-client resolution should go (as in S3)